server/client_test.go

```diff
@@ -456,6 +456,43 @@ func TestClientPubWithQueueSub(t *testing.T) {
 	}
 }
 
+func TestNonTLSRemoteAddress(t *testing.T) {
+	opts := DefaultOptions()
+	s := RunServer(opts)
+	defer s.Shutdown()
+
+	nc, err := nats.Connect(fmt.Sprintf("nats://%s:%d", opts.Host, opts.Port))
+	if err != nil {
+		t.Fatalf("Error on connect: %v", err)
+	}
+	defer nc.Close()
+
+	// Grab the client from server and set no echo by hand.
+	s.mu.Lock()
+	lc := len(s.clients)
+	c := s.clients[s.gcid]
+	s.mu.Unlock()
+
+	if lc != 1 {
+		t.Fatalf("Expected only 1 client but got %d\n", lc)
+	}
+	if c == nil {
+		t.Fatal("Expected to retrieve client\n")
+	}
+	c.mu.Lock()
+	c.echo = false
+	c.mu.Unlock()
+
+	addr := c.RemoteAddress()
+	if addr == nil {
+		t.Fatal("RemoteAddress() returned nil\n")
+	}
+
+	if addr.IP.String() != "127.0.0.1" {
```

Comment:
If we were going with `Addr`, this is where you would need to do `addr.(*net.TCPAddr).IP.String()`

Reply:
We already have pipes in the test which wouldnt have a `net.TCPAddr` hence these tests standing up servers and all - so kind of already hitting the scnario you worry about and I thought of it too.

For me, I generally hate it when some API returns a thing I then have to cast and test etc, sure we can return Addr, but then every consumer of this has to go and check what kind of Addr just like I had to do here between TLS and non TLS I find that's quite a unpleasant user API

Then i am almost more in favour of just returning a string `12.0.0.1:7562` or `pipe` or whatever

Reply:
Looking closer I see what you mean about Addr, guess I was thinking about something else. 

Let me take a stab st that tomorrow.

Reply:
The `net.Addr` suggestion was a good one thanks @kozlovic, I've redone this around that and I think its a lot better all round this way